pkg/controller/nodeipam/ipam/cidrset/cidr_set_test.go

```diff
@@ -736,3 +736,18 @@ func TestCIDRSetv6(t *testing.T) {
 		})
 	}
 }
+
+func benchmarkAllocateNextIPv6(subnetMaskSize int, b *testing.B) {
+	_, clusterCIDR, _ := net.ParseCIDR("2001:db8::/48")
+	a, _ := NewCIDRSet(clusterCIDR, subnetMaskSize)
+	for n := 0; n < b.N; n++ {
```

Comment:
The changes themselves look fine, though I don't think the benchmark is super-representative.
Basically what happens is that for large enough N:
- previously we were going over whole table to find a free bit
- now we exit early
I don't think it happens often in reality that we try to allocate many times if we don't have anything to allocate.

I'm fine with the improvement itself, but I think the better benchmark would be to in a loop:
- call maxCidrs time AllocateNext()
- the maxCidrs time Release() to free the usage

[currently the benchmark is pretty much proving that the corner case works fine only]